tokio/src/io/poll_evented.rs

```diff
@@ -166,7 +166,7 @@ feature! {
                     Ok(n) => {
                         // if we read a partially full buffer, this is sufficient on unix to show
                         // that the socket buffer has been drained
-                        if n > 0 && (!cfg!(windows) && n < len) {
+                        if n > 0 && (!cfg!(windows) && !cfg!(mio_unsupported_force_poll_poll) && n < len) {
```

Comment:
You should add some inline comments to these files explaining why that cfg is there.